cql3/statements/describe_statement.cc

```diff
@@ -0,0 +1,535 @@
+        , _slice(std::move(slice))
+        , _is_listing(is_listing)
+        , _selection(std::move(selection))
+        , _error_msg(std::move(error_msg)) {}
+
+    virtual ::shared_ptr<const metadata> get_result_metadata() const override {
+        return _selection->get_result_metadata();
+    }
+};
+
+namespace raw {
+
+describe_statement::describe_statement(describe_type desc_type)
+: _desc_type(desc_type)
+, _element_type(describe_statement::SCHEMA)
+, _element_keyspace(std::nullopt)
+, _element_name(std::nullopt)
+, _with_internals(false)
+, _is_listing(false)
+, _full_schema(false)
+, _only_keyspace(false) {}
+
+describe_statement::describe_statement(bool full_schema)
+: _desc_type(describe_type::schema)
+, _element_type(describe_statement::SCHEMA)
+, _element_keyspace(std::nullopt)
+, _element_name(std::nullopt)
+, _with_internals(false)
+, _is_listing(false)
+, _full_schema(full_schema)
+, _only_keyspace(false) {}
+
+describe_statement::describe_statement(std::optional<sstring> element_keyspace, bool only_keyspace)
+: _desc_type(describe_type::keyspace)
+, _element_type(describe_statement::KEYSPACE)
+, _element_keyspace(std::move(element_keyspace))
+, _element_name(std::nullopt)
+, _with_internals(false)
+, _is_listing(true)
+, _full_schema(false)
+, _only_keyspace(only_keyspace) {}
+
+describe_statement::describe_statement(describe_type desc_type, sstring element_type, std::optional<sstring> element_keyspace, std::optional<sstring> element_name, bool is_listing) 
+: _desc_type(desc_type)
+, _element_type(std::move(element_type))
+, _element_keyspace(std::move(element_keyspace))
+, _element_name(std::move(element_name))
+, _with_internals(false)
+, _is_listing(is_listing)
+, _full_schema(false)
+, _only_keyspace(false) {}
+
+void describe_statement::with_internals_details() {
+    _with_internals = true;
+}
+
+::shared_ptr<selection::selection> describe_statement::get_selection(schema_ptr schema) {
+    std::vector<const column_definition*> columns;
+    std::vector<lw_shared_ptr<column_specification>> metadata;
+
+<<<<<<< Updated upstream
+    if (_desc_type == describe_type::element || _desc_type == describe_type::schema || _desc_type == describe_type::keyspace) {
+=======
+    auto pks = schema->partition_key_columns();
+    auto cks = schema->clustering_key_columns();
+    std::transform(pks.begin(), pks.end(), std::back_inserter(columns), [](auto& c) { return &c; });
+    std::transform(pks.begin(), pks.end(), std::back_inserter(metadata), [](auto& c) { return c.column_specification; });
+    std::transform(cks.begin(), cks.end(), std::back_inserter(columns), [](auto& c) { return &c; });
+    std::transform(cks.begin(), cks.end(), std::back_inserter(metadata), [](auto& c) { return c.column_specification; });
+
+    if (_desc_type == describe_type::element || _desc_type == describe_type::schema || _desc_type == describe_type::keyspace || _desc_type == describe_type::generic) {
+>>>>>>> Stashed changes
```

Comment:
git marker left in code

Reply:
How did this even compile? :-)

Reply:
This is leftover from rebasing new changes. Only present in previos commit, not the last one, so it was compiling.
Fixed :)